app/src/main/java/com/example/tasky/authentication/domain/usecase/ValidateEmailUseCase.kt

```diff
@@ -0,0 +1,15 @@
+package com.example.tasky.authentication.domain.usecase
+
+import com.example.tasky.authentication.domain.EmailValidator
+import javax.inject.Inject
+
+interface ValidateEmailUseCase {
```

Comment:
I believe its a better practice to depend on interfaces, its better for testing and maybe in some case in the future the implementation of the use case could change and if the viewmodel depends on the interface won't be affected by the change of implementation. What do you think @philipplackner ?

Reply:
Yes, interfaces are great for stuff like a Repository, or to avoid having android dependencies on domain . Ex: EmailMatcher

But for usecases. Whats the benefit? When testing, its fine to use the real usecase. Since that's what you want to test